src/lib/hooks/useTokenList/fetchTokenList.ts

```diff
@@ -43,31 +43,38 @@ export default async function fetchTokenList(
     urls = uriToHttp(listUrl)
   }
 
+  if (urls.length === 0) {
+    throw new Error('Unrecognized list URL protocol.')
+  }
+
   for (let i = 0; i < urls.length; i++) {
     const url = urls[i]
-    const isLast = i === urls.length - 1
     let response
     try {
       response = await fetch(url, { credentials: 'omit' })
     } catch (error) {
       const message = `failed to fetch list: ${listUrl}`
       console.debug(message, error)
-      if (isLast) throw new Error(message)
       continue
     }
 
     if (!response.ok) {
       const message = `failed to fetch list: ${listUrl}`
       console.debug(message, response.statusText)
-      if (isLast) throw new Error(message)
       continue
     }
 
-    const json = await response.json()
-    const list = skipValidation ? json : await validateTokenList(json)
-    listCache?.set(listUrl, list)
-    return list
+    try {
+      const json = await response.json()
```

Comment:
If the response status was `ok`, when can this be an invalid json? Do we know that? I think it would be great to include that as a comment, b/c it's pretty confusing right now.

Reply:
the main failure case i'm trying to catch here is the one where `validateTokenList` fails, not when the `response.json()` itself is not parsable.